.eslintrc.js

```diff
@@ -44,10 +44,19 @@ module.exports = {
         "indent": ["warn", 4, {"SwitchCase":1}],
         "linebreak-style": ["error", "unix"],
         "semi": ["error", "always"],
+        "keyword-spacing": ["error"],
+        "no-implicit-globals": ["error"],
+        "no-irregular-whitespace": ["error"],
         "no-new-object": ["error"],
-        "no-unneeded-ternary": ["error"],
+        "no-regex-spaces": ["error"],
+        "no-spaced-func": ["error"],
         "no-throw-literal": ["error"],
-        "no-implicit-globals": ["error"],
+        "no-unneeded-ternary": ["error"],
+        "no-whitespace-before-property": ["error"], // match flake8 E201 and E211
+        "space-before-function-paren": ["error", "never"],
```

Comment:
I think the rule we have been following (and what I prefer and picked up from crockford style in javascript the good parts when we started trying to have conventions) is `"space-before-function-paren": ["error", {"anonymous": "always", "named": "never", "asyncArrow": "always"}]` meaning

```
function () { // space before anonymous function paren
}
```
but
```
function foo() {  // no space before named function paren
}
```

Reply:
(offlined, going to implement Danny's suggestion)